--- antlr-2.7.6/configure.in.orig	2005-12-07 19:00:32.000000000 +0100
+++ antlr-2.7.6/configure.in	2006-05-22 22:20:47.302510750 +0200
@@ -944,8 +944,8 @@
     )
 
     AC_CONFIG_FILES([lib/csharp/Makefile])
-    AC_CONFIG_FILES([lib/csharp/src/Makefile])
-    AC_CONFIG_FILES([lib/csharp/ASTFrame/Makefile])
+    AC_CONFIG_FILES([lib/csharp/antlr.runtime/Makefile])
+    AC_CONFIG_FILES([lib/csharp/antlr.astframe/Makefile])
 
     # We have a Makefile that loops through all python examples.
     case $WITH_EXAMPLES in
--- antlr-2.7.6/scripts/config.deps.in.orig	2005-12-07 19:00:36.000000000 +0100
+++ antlr-2.7.6/scripts/config.deps.in	2006-05-22 22:34:53.887419000 +0200
@@ -109,12 +109,12 @@
 ## double colon rule, ie. further  rules  with the same target may be
 ## added. For unknown reasons, double colon rules are always phony,ie.
 ## getting executed even in case target exists. We break the infinite
-## loop,  we only  jump  into  subdir  "lib/csharp/src" if we are not 
+## loop,  we only  jump  into  subdir  "lib/csharp/antlr.runtime" if we are not 
 ## already in. It is very important that  each Makefile[.in] sets the
 ## variable $(subdir) proper.
 
 @ANTLR_NET@ :: 
-	@ subdir="lib/csharp/src" ; \
+	@ subdir="lib/csharp/antlr.runtime" ; \
     case $(subdir) in \
 	  $$subdir ) ;; \
 	  * ) d="@abs_this_builddir@/$$subdir" ; \
@@ -125,7 +125,7 @@
     esac
 
 @ASTFRAME_NET@ :: 
-	@ subdir="lib/csharp/ASTFrame" ; \
+	@ subdir="lib/csharp/antlr.astframe" ; \
     case $(subdir) in \
 	  $$subdir ) ;; \
 	  * ) d="@abs_this_builddir@/$$subdir" ; \
