--- antlr-20030911/Makefile~	2003-09-11 15:04:10.000000000 +0200
+++ antlr-20030911/Makefile	2003-10-09 19:04:51.000000000 +0200
@@ -235,8 +235,6 @@
 	antlr/debug/TraceEvent.class \
 	antlr/debug/TraceListener.class \
 	antlr/debug/Tracer.class \
-	antlr/debug/misc/ASTFrame$$1.class \
-	antlr/debug/misc/ASTFrame$$MyTreeSelectionListener.class \
 	antlr/debug/misc/ASTFrame.class \
 	antlr/debug/misc/JTreeASTModel.class \
 	antlr/debug/misc/JTreeASTPanel.class \
