diff -uNr antlr-2.7.7.orig/antlr/Makefile.in antlr-2.7.7/antlr/Makefile.in
--- antlr-2.7.7.orig/antlr/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/antlr/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -321,11 +321,11 @@
 
 ## install antlr.jar in libdirectory if exists
 install ::	@ANTLR_JAR@ 
-	-$(MKDIR) -p "$(libdir)"
-	-$(MKDIR) -p "$(datadir)/$(versioneddir)"
+	-$(MKDIR) -p "$(DESTDIR)$(libdir)"
+	-$(MKDIR) -p "$(DESTDIR)$(datadir)/$(versioneddir)"
 	-test -f @ANTLR_JAR@ && {\
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(datadir)/$(versioneddir)" ; \
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(libdir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(libdir)" ; \
 	}
 
 distclean :: clean
diff -uNr antlr-2.7.7.orig/doc/Makefile.in antlr-2.7.7/doc/Makefile.in
--- antlr-2.7.7.orig/doc/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/doc/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -43,12 +43,12 @@
 	$(eol)
 
 install ::
-	@$(MKDIR) -p "$(antlr_doc_DIR)"
+	@$(MKDIR) -p "$(DESTDIR)$(antlr_doc_DIR)"
 	@@ECHO@ "install doc files .. "
 	@for f in $(antlr_doc_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_doc_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_doc_DIR)" ; \
 		fi ;\
 	done
 
diff -uNr antlr-2.7.7.orig/lib/cpp/antlr/Makefile.in antlr-2.7.7/lib/cpp/antlr/Makefile.in
--- antlr-2.7.7.orig/lib/cpp/antlr/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/lib/cpp/antlr/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -75,11 +75,11 @@
 
 install:
 	@@ECHO@ "install hpp files .. "
-	@$(MKDIR) -p "$(includedir)/antlr"
+	@$(MKDIR) -p "$(DESTDIR)$(includedir)/antlr"
 	@for f in $(antlr_hpp_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(includedir)/antlr" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(includedir)/antlr" ; \
 		fi ;\
 	done
 
diff -uNr antlr-2.7.7.orig/lib/cpp/src/Makefile.in antlr-2.7.7/lib/cpp/src/Makefile.in
--- antlr-2.7.7.orig/lib/cpp/src/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/lib/cpp/src/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -114,12 +114,12 @@
   $(eol)
 
 this-install: @ANTLR_LIB@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C++ core files .. "
 	@for f in $(antlr_lib_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 
diff -uNr antlr-2.7.7.orig/lib/csharp/antlr.astframe/Makefile.in antlr-2.7.7/lib/csharp/antlr.astframe/Makefile.in
--- antlr-2.7.7.orig/lib/csharp/antlr.astframe/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/lib/csharp/antlr.astframe/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -45,13 +45,13 @@
 
 ## install our target ..
 install :: @ASTFRAME_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C# core files .. "
 	@for f in @ASTFRAME_NET@  ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
diff -uNr antlr-2.7.7.orig/lib/csharp/antlr.runtime/Makefile.in antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in
--- antlr-2.7.7.orig/lib/csharp/antlr.runtime/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in	2006-11-07 23:22:25.147461999 +0100
@@ -135,13 +135,13 @@
 	@RMF@ Makefile
 
 install :: @ANTLR_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install C# core files .. "
 	@for f in @ANTLR_NET@  ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
diff -uNr antlr-2.7.7.orig/lib/csharp/antlr.runtime/Makefile.in~ antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in~
--- antlr-2.7.7.orig/lib/csharp/antlr.runtime/Makefile.in~	1970-01-01 01:00:00.000000000 +0100
+++ antlr-2.7.7/lib/csharp/antlr.runtime/Makefile.in~	2006-11-01 22:37:17.000000000 +0100
@@ -0,0 +1,153 @@
+##xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+@stdvars@
+##xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+## do not change this value 
+subdir=lib/csharp/antlr.runtime
+
+all :: lib
+
+lib: @ANTLR_NET@
+
+antlr_cs_FILES = \
+  @abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ANTLRException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ANTLRPanicException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ASTFactory.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ASTNodeCreator.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ASTNULLType.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ASTPair.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ASTVisitor.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/BaseAST.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ByteBuffer.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CharBuffer.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CharQueue.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CharScanner.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CharStreamException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CharStreamIOException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CommonAST.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CommonASTWithHiddenTokens.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CommonHiddenStreamToken.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/CommonToken.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/DefaultFileLineFormatter.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/DumpASTVisitor.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/FileLineFormatter.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/IHiddenStreamToken.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/InputBuffer.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/IToken.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/LexerSharedInputState.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/LLkParser.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/MismatchedCharException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/MismatchedTokenException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/NoViableAltException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/NoViableAltForCharException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/Parser.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ParserSharedInputState.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ParseTree.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ParseTreeRule.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/ParseTreeToken.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/RecognitionException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/SemanticException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/StringUtils.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenBuffer.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/Token.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenCreator.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenQueue.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamBasicFilter.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStream.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamHiddenTokenFilter.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamIOException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamRecognitionException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamRetryException.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamRewriteEngine.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenStreamSelector.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TokenWithIndex.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TreeParser.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr/TreeParserSharedInputState.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.collections/AST.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.collections.impl/ASTArray.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.collections.impl/BitSet.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ANTLREventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/DebuggingCharScanner.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/DebuggingInputBuffer.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/DebuggingParser.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/GuessingEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ICharScannerDebugSubject.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/IDebugSubject.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/InputBufferEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/InputBufferEventSupport.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/InputBufferListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/InputBufferListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/InputBufferReporter.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/IParserDebugSubject.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/Listener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/LLkDebuggingParser.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/MessageEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/MessageListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/MessageListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/NewLineEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/NewLineListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserController.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserEventSupport.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserMatchEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserMatchListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserMatchListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserReporter.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserTokenEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserTokenListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParserTokenListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ParseTreeDebugParser.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/ScannerEventSupport.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/SemanticPredicateEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/SemanticPredicateListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/SemanticPredicateListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/SyntacticPredicateEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/SyntacticPredicateListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/SyntacticPredicateListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/TraceEventArgs.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/TraceListenerBase.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/TraceListener.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/antlr.debug/Tracer.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/SupportClass.cs\
+	@abs_top_srcdir@/lib/csharp/antlr.runtime/AssemblyInfo.cs\
+  $(eof)
+
+
+clean-target ::
+	@ @RMF@ @ANTLR_NET@
+
+@ANTLR_NET@ :: $(antlr_cs_FILES) 
+	 @- @RMF@ $@
+	 @  @CSHARP_COMPILE_CMD@  $@ $(antlr_cs_FILES)
+	 @  test -f  $@ || exit 1
+
+clean ::
+	@RMF@ *.obj *.o *.a  *.lib *.so *.dll *~ @ANTLR_NET@
+
+## use this target if you just want to rebuild the lib without
+## compiling again.
+clean-lib :
+	@RMF@ @ANTLR_NET@
+
+distclean :: clean
+	@RMF@ Makefile
+
+install :: @ANTLR_NET@ 
+	@$(MKDIR) -p "$(libdir)"
+	@@ECHO@ "install C# core files .. "
+	@for f in @ANTLR_NET@  ; do \
+		@ECHO@ "install $${f}" ; \
+		if test -f "$${f}" ; then \
+			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+		fi ;\
+	done
+
+## dependencies
+@ANTLR_NET@        :: Makefile
+@ANTLR_NET@        :: @abs_this_builddir@/scripts/csc.sh
+
+## other dependencies to be listed below
+@stddeps@
diff -uNr antlr-2.7.7.orig/lib/python/Makefile.in antlr-2.7.7/lib/python/Makefile.in
--- antlr-2.7.7.orig/lib/python/Makefile.in	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7/lib/python/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -41,26 +41,26 @@
 extradir = $(datadir)/$(versioneddir)
 
 install-this:
-	$(MKDIR) -p "$(antlr_py_DIR)"
-	$(MKDIR) -p "$(libdir)"
+	$(MKDIR) -p "$(DESTDIR)$(antlr_py_DIR)"
+	$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@@ECHO@ "install python files .. "
 	@for f in $(antlr_py_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 	@for f in $(antlr_py1_FILES) ; do \
 		@ECHO@ "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_py_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_py_DIR)" ; \
 		fi ;\
 	done
 	@f="$(objdir)/scripts/pyantlr.sh"; \
 	if test -f "$${f}" ; then \
 			@ECHO@ "install Python installer" ; \
-			$(MKDIR) -p "$(sbindir)" ; \
-			$(INSTALL) -m 555 "$${f}" "$(sbindir)" ; \
+			$(MKDIR) -p "$(DESTDIR)$(sbindir)" ; \
+			$(INSTALL) -m 555 "$${f}" "$(DESTDIR)$(sbindir)" ; \
 	fi
 
 
diff -uNr antlr-2.7.7.orig/Makefile.in antlr-2.7.7/Makefile.in
--- antlr-2.7.7.orig/Makefile.in	2006-11-01 22:37:18.000000000 +0100
+++ antlr-2.7.7/Makefile.in	2006-11-07 23:21:49.133211249 +0100
@@ -124,16 +124,16 @@
 extradir = $(datadir)/$(versioneddir)
 
 install ::
-	$(MKDIR) -p "$(bindir)"
-	$(MKDIR) -p "$(extradir)"
-	$(MKDIR) -p "$(docdir)"
-	$(INSTALL) -m 755 scripts/run-antlr                          "$(bindir)/antlr"
-	$(INSTALL) -m 755 scripts/antlr-config                       "$(bindir)/antlr-config"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(docdir)"
+	$(MKDIR) -p "$(DESTDIR)$(bindir)"
+	$(MKDIR) -p "$(DESTDIR)$(extradir)"
+	$(MKDIR) -p "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 755 scripts/run-antlr                          "$(DESTDIR)$(bindir)/antlr"
+	$(INSTALL) -m 755 scripts/antlr-config                       "$(DESTDIR)$(bindir)/antlr-config"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(DESTDIR)$(docdir)"
 
 install ::
 	@ECHO@ "installation done"
