--- antlr-2.7.6/antlr/Makefile.in.orig	2006-08-03 23:57:11.142314000 +0200
+++ antlr-2.7.6/antlr/Makefile.in	2006-08-03 23:57:27.343326500 +0200
@@ -133,6 +133,9 @@
   @abs_top_srcdir@/antlr/JavaBlockFinishingInfo.java \
   @abs_top_srcdir@/antlr/JavaCharFormatter.java \
   @abs_top_srcdir@/antlr/JavaCodeGenerator.java \
+  @abs_top_srcdir@/antlr/JavaCodeGeneratorPrintWriterManager.java \
+  @abs_top_srcdir@/antlr/DefaultJavaCodeGeneratorPrintWriterManager.java \
+  @abs_top_srcdir@/antlr/PrintWriterWithSMAP.java \
   @abs_top_srcdir@/antlr/LLkAnalyzer.java \
   @abs_top_srcdir@/antlr/LLkGrammarAnalyzer.java \
   @abs_top_srcdir@/antlr/LLkParser.java \
@@ -192,6 +193,7 @@
   @abs_top_srcdir@/antlr/TreeParserSharedInputState.java \
   @abs_top_srcdir@/antlr/TreeSpecifierNode.java \
   @abs_top_srcdir@/antlr/TreeWalkerGrammar.java \
+  @abs_top_srcdir@/antlr/Utils.java \
   @abs_top_srcdir@/antlr/WildcardElement.java \
   @abs_top_srcdir@/antlr/ZeroOrMoreBlock.java \
   @abs_top_srcdir@/antlr/build/ANTLR.java \
